### PR TITLE
Enable NMS for multiple output

### PR DESCRIPTION
**Pull request by kelvinchoi-intel**

### CODE REVIEW

**Review by ahnyoung-paul — APPROVED** [review]
LGTM

File: src/plugins/intel_gpu/src/graph/program_node.cpp [hunk 1]
```diff
@@ -33,7 +33,8 @@ program_node::program_node(std::shared_ptr<primitive> prim, program& prog)
         num_outputs = prim->num_outputs;
         for (int i = 0 ; i < num_outputs; ++i) {
             layout output_layout = layout(data_types::f32, format::bfyx, tensor());
-            output_layout.data_padding = prim->output_paddings[i];
+            // TODO(kelvin): Apply each output padding by output index. ex) output_layout.data_padding = prim->output_paddings[i];
+            output_layout.data_padding = prim->output_paddings[0];
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/program_node.cpp:37]: why is this changed?"

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/non_max_suppression/non_max_suppression_kernel_ref.cpp [hunk 2]
```diff
@@ -224,10 +227,16 @@ void NonMaxSuppressionKernelRef::SetKernelArguments(const non_max_suppression_pa
         kernel.params.arguments.push_back({ ArgumentDescriptor::Types::INTERNAL_BUFFER, 1 });
         kernel.params.arguments.push_back({ ArgumentDescriptor::Types::INTERNAL_BUFFER, 0 });
 
+        // TODO(kelvin)
+        #if 0
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/non_max_suppression/non_max_suppression_kernel_ref.cpp:231]: please remove unused codes

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/non_max_suppression/non_max_suppression_kernel_ref.h [hunk 3]
```diff
@@ -63,6 +60,8 @@ struct non_max_suppression_params : public base_params {
         return input_idx;
     }
 
+    // TODO(kelvin)
+    #if 0
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/non_max_suppression/non_max_suppression_kernel_ref.h:64]: please remove unused code

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/non_max_suppression/non_max_suppression_kernel_ref.cpp (hunk 2 above)
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/non_max_suppression/non_max_suppression_kernel_ref.cpp:231]: fixed

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/non_max_suppression/non_max_suppression_kernel_ref.h (hunk 3 above)
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/non_max_suppression/non_max_suppression_kernel_ref.h]: fixed

File: src/plugins/intel_gpu/src/graph/program_node.cpp (hunk 1 above)
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/src/graph/program_node.cpp:37]: Multiple paddings for multiple output should be implemented but right now 0 index for all output_paddings is enough. I'd like to update this part at later.
